packages/webview_flutter/webview_flutter_android/lib/src/android_webview_controller.dart

```diff
@@ -740,6 +750,16 @@ class AndroidNavigationDelegate extends PlatformNavigationDelegate {
           callback(url);
         }
       },
+      onReceivedHttpError: (
+        android_webview.WebView webView,
+        android_webview.WebResourceRequest request,
+        android_webview.WebResourceResponse response,
+      ) {
+        if (weakThis.target?._onHttpError != null) {
+          weakThis.target!._onHttpError!(
+              HttpResponseError(statusCode: response.statusCode));
```

Comment:
I'm confused; how is this callback intended to work in practice? You are throwing away the request object before it gets to the client, which means they will have absolutely no way of knowing whether the error is coming from a navigation (and if so to where) or a resource on the page (and if so, what). I'm not clear on what non-trivial use case a client could satisfy without having the request.

Am I missing something here?

Reply:
/cc @bparrishMines who reviewed the overall PR, since it's core to the overall design. (It looks like the main PR only had one review before sub-parts started landing? Ideally that should have been two.)

Reply:
It makes sense to include the request in the HttpResponseError object. Is it better to create a platform specific implementation for HttpResponseError and include the request in there, like AndroidWebResourceError, or include it in the HttpResponseError object itself?

Reply:
Since it seems core to using this correctly, I would include it in the base object; it could be nullable in case there are implementations or cases where it's not available.

Reply:
@stuartmorgan Sorry about that. I've been reviewing all of the WebView PRs as the owner of both the Android and iOS implementation, so I assumed it was fine to be the only approval for the aggregate PR. I'll start getting secondary reviews for those as well.

I agree with creating adding a request object to `HttpResponseError`.

Reply:
It actually looks like Android and iOS have a request object and a response object. 

Android:
https://developer.android.com/reference/android/webkit/WebResourceRequest
https://developer.android.com/reference/android/webkit/WebResourceResponse

iOS:
https://developer.apple.com/documentation/foundation/nsurlrequest?language=objc
https://developer.apple.com/documentation/foundation/nshttpurlresponse

In their respective methods,
https://developer.apple.com/documentation/webkit/wknavigationdelegate/1455643-webview?language=objc
https://developer.android.com/reference/android/webkit/WebViewClient#onReceivedHttpError(android.webkit.WebView,%20android.webkit.WebResourceRequest,%20android.webkit.WebResourceResponse)

Android provides the request and response and iOS provides the response. I think we could add a nullable `LoadRequestParams` to `HttpResponseError` AND a new nullable `WebResourceResponse` field.